pkg/deploy/util/util.go

```diff
@@ -425,9 +416,9 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	} else {
 		// if there is an active deployment, we need to filter out any deployments that we don't
 		// care about, namely the active deployment and any newer deployments
-		for _, deployment := range deployments {
-			if &deployment != activeDeployment || DeploymentVersionFor(&deployment) < DeploymentVersionFor(activeDeployment) {
-				relevantDeployments = append(relevantDeployments, deployment)
+		for i := range deployments {
+			if &deployments[i] != activeDeployment && DeploymentVersionFor(&deployments[i]) < DeploymentVersionFor(activeDeployment) {
```

Comment:
@kargakis In “ActiveDeployment” compare the "deploymentVersion" should return the latest complete deployment and not change the "input". Thanks.